refactor(codegen): clarify choice of quote when printing strings

overlookmotel opened the pull request:
Pure refactor. Make the logic for choice of quote character when printing strings a bit clearer.

Boshen commented:
### Merge activity

* **Aug 31, 5:20 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(codegen): clarify choice of quote when printing strings (#13440)

Pure refactor. Make the logic for choice of quote character when printing strings a bit clearer.
